csunplugged/templates/topics/programming-challenges-table.html

```diff
@@ -2,41 +2,100 @@
+            {% trans "Show more details" %}
+          </span>
+        </td>
+        <td class="text-center">
+          {% include "topics/programming-difficulty-badge.html" %}
+        </td>
+        <td class="text-center">
+          {% for implementation in programming_challenge.implementations.all %}
+            <img src="{% get_static_prefix %}{{ implementation.language.icon }}" class="inline-image-small" />
+          {% endfor %}
+        </td>
+      </tr>
+      <tr>
+        <td class="p-0"></td>
+        <td colspan="3"  class="p-0">
+            <div id="accordion-{{ programming_challenge.challenge_set_number }}{{ programming_challenge.challenge_number }}" class="collapse acc-programming-challenge">
+              {% if programming_challenge.learning_outcomes %}
+                <strong>{% trans "Learning outcomes" %}</strong>
+                <div class="boxed-text-content">
+                  <p>{% trans "Students will be able to:" %}</p>
+                  <ul>
+                    {% for learning_outcome in programming_challenge.learning_outcomes.all %}
+                    <li>
+                      {{ learning_outcome.text }}<br>
+                      {% for area in learning_outcome.curriculum_areas.all %}
+                        {% include "topics/curriculum-area-badge.html" %}
+                      {% endfor %}
+                    </li>
+                    {% endfor %}
+                  </ul>
+                </div>
+              {% endif %}
+
+              {% if programming_challenge.implementations %}
+                <strong>{% trans "Challenge Solutions" %}</strong>
+              {% endif %}
+              {% for implementation in programming_challenge.implementations.all %}
+                </br>
+                <a href="{% url 'topics:programming_challenge_solution' topic.slug programming_challenge.slug implementation.language.slug %}">
+                    {% blocktrans with name=implementation.language.name %}Show {{ name }} solution{% endblocktrans %}
+                </a> 
+              {% endfor %}
+            </div>
+        </td>
+      </tr>
+      {% endfor %}
+    </tbody>
+  </table>
+</div>
+
+{% block scripts %}
+  <script type="text/javascript">
+    function updateToggleButtonText(challenge_num) {
+      challengeContentToggleId = `#toggle-${challenge_num}`;
+      if ($(challengeContentToggleId).hasClass('collapsed')) {
+        $(challengeContentToggleId).text('Show less details');
+      } else {
+        $(challengeContentToggleId).text('Show more details');
```

Comment:
If two challenges have the same ID number (see [How Binary Digits Work](https://cs-unplugged-dev.appspot.com/en/topics/binary-numbers/unit-plan/how-binary-digits-work/programming/)) Both sets of outcomes are expanded on one button push

Reply:
Awesome spot